### PR TITLE
[MAINTENANCE] Add `add_expectation_suite` to `DataContext` CRUD

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Add `add_expectation_suite` and related tests
- Misc cleanup

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 1]
```diff
@@ -2140,22 +2142,105 @@ def create_expectation_suite(
...
+        return self._add_expectation_suite(
+            expectation_suite_name=expectation_suite_name,
+            id=id,
+            expectations=expectations,
+            evaluation_parameters=evaluation_parameters,
+            data_asset_type=data_asset_type,
+            execution_engine_type=execution_engine_type,
+            meta=meta,
+            overwrite_existing=False,  # `add` does not resolve collisions
+        )
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/abstract_data_context.py:2208]: It looks like the public method is just a wrapper around the private method but all it does is call the private method.

Can we simplify this?
Get rid of the private method and put all the logic in the public method?

Comment (cdkini) [reply great_expectations/data_context/data_context/abstract_data_context.py:2208]: `overwrite_existing` is an outdated parameter we want to move away from (it is used in the legacy `create_expectation_suite`) - we can get rid of this private method once we deprecate and remove `create`

Comment (cdkini) [reply great_expectations/data_context/data_context/abstract_data_context.py:2208]: note that this private method is used in both public methods

Comment (Kilo59) [reply great_expectations/data_context/data_context/abstract_data_context.py:2208]: Oh I see, I missed that it's being used in 2 places.

**Review by Kilo59 — APPROVED** [review]

**Pull request merged**